lib/gen_socket_client/serializer.ex

```diff
@@ -51,7 +51,7 @@ defmodule Phoenix.Channels.GenSocketClient.Serializer.GzipJson do
   @doc false
   def encode_message(message) do
     case Jason.encode_to_iodata(message) do
-      {:ok, encoded} -> {:ok, {:binary, :zlib.gzip(encoded)}}
+      {:ok, encoded} -> {:ok, {:text, :zlib.gzip(encoded)}}
```

Comment:
Is Phoenix supposed to handle the case where the `:text` payload is gzipped? If possible, could you try using `Serializer.GzipJson` as the serializer and see if it works?

Reply:
You are right, out of the box Phoenix doesn't handle Gziped case, channel join silently fails. But in theory it could work if you provide your own serializer on the Phoenix side. So probably it should be left as binary as it was before, or be removed.

Reply:
> So probably it should be left as binary as it was before, or be removed.

I fear `:binary` won't work either, because technically this isn't a binary - it's a JSON string. Elixir adds to the confusion by referring to strings as `binary`, which is probably the reason for this bug in the first place.

I'm not sure what to do with this. This change is not productive if Phoenix does not accept it out of the box. We can leave it as it was and then consider dropping the module entirely if it's not being used by anyone.

Any thoughts?

Reply:
Let's fix only the regular (non gzip) module and make a minor release.